### ISSUE TITLE
Update Template from SD2.0.0 to SD3.0.0 

### ISSUE THREAD

**Issue by rgleason** [Rasbats/shipdriver_pi#357]
I have been reading the manual trying to figure out what is necessary to update VDR_pi CI template which I had working with a previous version.

VDR_pi is building locally now and it works. The next step is to update permissions I think.
I have already made a key for logbook and had it accepted and logbook works ok as it stands, not updated (that is next at some point)

Is this where I should start really following the manual?
https://opencpn-manuals.github.io/main/AlternativeWorkflow/InstallConfigure/Cloud-Service-Changes.html#_experimenting_with_a_new_build

### Fix File permissions due to Windows
and do this to fix permissions?

```
    git update-index –chmod=+x foo.sh
    git commit -m ``Adding execute permissions to foo.sh''
    git push

```

### Make new Credentials for the specific plugin
When creating credentials for Logbookkonni I believe that was necessary in order to even build on circleci. Now it seems it is just necessary to be able to automatically push to plugins.  Is that correct?

### What else do I need to do to be able to push a release?

**Comment by rgleason** [comment]
Also can I use the same SSH as I used for Logbook?

**Comment by leamas** [comment]
BTW: The manual is definitely outdated in this area. The canonical way to update to sd3.0.0 is to use `update-templates sd3.0.0`, possibly after first bootstrapping update-templates. See UPDATE_TEMPLATES.md.

**Comment by Rasbats** [comment]
>See UPDATE_TEMPLATES.md

Finally worked out what [treeish] means. Honest! Needs an example in UPDATE_TEMPLATES.md.

**Comment by leamas** [comment]
Right, updates under way...

**Comment by leamas** [comment]
This is what I have right now. Is it ok? 

```
Usage summary:

    update-templates [-T] [treeish]
    update-templates  -h
    update-templates  -l

**treeish** defaults to _shipdriver/master_ i. e., templates are updated
from shipdriver's development branch. It could be set to a branch
like _shipdriver/v3.0_ or a tag like _sd3.0.0_ to retrieve data from
corresponding git trees.
```

**Comment by leamas** [comment]
OK, let's add an Examples section as well:

```
Examples:

    update-templates shipdriver/v3.0    -- get updates form v3.0 branch
    update-templates sd3.0.0            -- get updates from sd3.0.0 tag
    update-templates -l                 -- list all available tags
```
That should fill in the missing gaps, I guess. Or?

**Comment by Rasbats** [comment]
Good. Thanks.

**Comment by leamas** [comment]
Submitted in #359

**Comment by leamas** [comment]
What is the issue here now? We know the manual is outdated, thats added to https://github.com/Rasbats/managed_plugins/issues/24.

Obviously, UPDATE_TEMPLATES.md also needed some polish, but this is now in place.

Given all this, what is this issue about now?

**Comment by rgleason** [comment]
I did not realize I could simply execute something to update the SD Template. Quite frankly, I get a bit lost in the TOC and have trouble finding relevant things. I know it is difficult to keep up with the documentation when development of the template has not matured. Perhaps next cycle I will use the update_templates feature, but for now, I did it the way I do Testplugin using WinMerge

**Comment by leamas** [comment]
>  did it the way I do Testplugin using WinMerge

which is unsupported, and always will fail.  So, if the issue is that you cannot update the templates using WinMerge, the issue is invalid. 

You are of course free to update things whichever way you want. But requiring support when doing so is problematic,

**Comment by rgleason** [comment]
I have closed this.

**Comment by rgleason** [comment]
I wrote earlier:

`I did not realize I could simply execute something to update the SD Template. Quite frankly, I get a bit lost in the TOC and have trouble finding relevant things. I know it is difficult to keep up with the documentation when development of the template has not matured. Perhaps next cycle I will use the update_templates feature, but for now, I did it the way I do Testplugin using WinMerge`

And before that I listed two things that I thought I needed to do to update vdr_pi from earlier SD Template and asked if there was anything I had missed.
